### PR TITLE
Speed up SPM CI by reducing full Firebase rebuilds

### PR DESCRIPTION
**Pull request by paulb777**
#no-changelog

### CODE REVIEW

**Review by ryanwilson — APPROVED** [review]
Looks great! What kind of difference do we see with this?

File: .github/workflows/spm.yml [hunk 1]
```diff
@@ -48,10 +48,13 @@ jobs:
       matrix:
         target: [tvOS, macOS, catalyst]
         # Full set of Firebase-Package tests only run on iOS because of Analytics.
```
Comment (ryanwilson) [inline .github/workflows/spm.yml:50]: Nit: comment should go above now? or not relevant anymore?

**Review by ryanwilson — APPROVED** [review]

File: .github/workflows/spm.yml (hunk 1 above)
Comment (paulb777) [reply .github/workflows/spm.yml:50]: Still relevant

**Comment by paulb777** [comment]
We go from three jobs per platform - see https://github.com/firebase/firebase-ios-sdk/runs/2800987939?check_suite_focus=true to one job per platform - see https://github.com/firebase/firebase-ios-sdk/actions/runs/926785832

tvos goes from 27 cpu minutes to 9
macos 23 to 14
catalyst 26 to 9

**Pull request merged**